packages/tutanota-error/lib/TutanotaError.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * Base class for all errors in Tutanota. Provides the handling of error stacks for chrome (captureStackTrace) and others.
+ * Implemented using ES5 inheritance as babel does not support extending builtin types
+ * @see http://stackoverflow.com/questions/33870684/why-doesnt-instanceof-work-on-instances-of-error-subclasses-under-babel-node
+ *
+ * Note that passing errors between worker and main thread requires all fields of the error to be serializable.
+ * Currently those are name, message, stack and data. See (errorToObj() and objToError()).
+ *
+ * In order to correctly set the class type of the error after deserialization
+ * (needed for e instanceof CustomError to work), the error class needs to be
+ * added to the ErrorNameToType map in Utils.js.
+ */
+const ExtendableErrorF = function ExtendableError() {
+	// @ts-ignore
+	Error.apply(this, arguments)
+}
+
+// Warning: huge type hack
+// You can't import downcast here
+ExtendableErrorF.prototype = Object.create(Error.prototype)
+// @ts-ignore
+const ExtendableError: Class<Error> = ExtendableErrorF as any
+
+export class TutanotaError extends ExtendableError {
+	name: string
+	message: string
+
+	constructor(name: string, message: string) {
+		super(message)
+		this.name = name
+		this.message = message
+
+		if (typeof Error.captureStackTrace === "function") {
+			Error.captureStackTrace(this, this.constructor)
+		} else {
+			let error = new Error()
```

Comment:
can we assume that stacks are always set by now? at least IE9 should be irrelevant today right?

Reply:
I suspect it's more than IE9 and I didn't want to take this risk